File: problem/digit-palindrome/6/main.cs
// main.cs: main file for solutions to the digit-palindrome exercise
//
// This is to be linked to the submitted file.
// It processes values from standard input then from the "in.txt" file.
//
//
// Copyright (C) 2020-2022  Rudy Matela
// All rights reserved.

using System;

class TheNewMain
{
	static void Main()
	{
		string line;

		// processes standard input
		while ((line = Console.ReadLine()) != null) {
			Solve1(line);
		}

		// processes additional inputs from in.txt
		var file = new System.IO.StreamReader("in.txt");
		while ((line = file.ReadLine()) != null) {
			Solve1(line);
		}
		file.Close();
	}

	static void Solve1(string line) {
		int x = Convert.ToInt32(line);
		if (Program.Palindrome(x))
			Console.WriteLine($"{x} is palindrome");
		else
			Console.WriteLine($"{x} is not palindrome");
	}
}

File: problem/digit-palindrome/6/main.c
/*
 * main.c: main file for a solution to the palindrome exercise
 *
 * This is to be linked to the submitted file.
 * It processes values from standard input then from the "in.txt" file.
 *
 *
 * Copyright (C) 2020-2024  Rudy Matela
 * All rights reserved.
 */
#include <stdio.h>
int palindrome(int);
int main()
{
	int i, j;
	FILE *f;
	/* process standard input */
	while (scanf(" %d", &i)==1)
		if (palindrome(i))
			printf("%d is palindrome\n", i);
		else
			printf("%d is not palindrome\n", i);
	/* process additional inputs from in.txt */
	f = fopen("in.txt", "r");
	if (!f) {
		fprintf(stderr, "could not open in.txt");
		return 1;
	}
	while (fscanf(f," %d", &i)==1)
		if (palindrome(i))
			printf("%d is palindrome\n", i);
		else
			printf("%d is not palindrome\n", i);
	fclose(f);
	/* in Python, we cannot load a library without "running" it.
	 * so for this test case, we provide additional inputs on "in.txt"
	 * cf. main.py
	 */
	return 0;
}
